lib/logstash/inputs/google_pubsub.rb

```diff
@@ -249,7 +249,7 @@ def run(queue)
         @logger.error("Error pulling messages:'#{result.error_message}'")
       end
 
-      if messages
+      if messages.any?
```

Comment:
Unfortunately, it’s entirely possible for `messages` to be `nil`, if the payload doesn’t include a value fit that key. In this case, we would get a `NoMethodError` when attempting to send `Enumerable#any?` to `nil`; the following would be safer:

~~~ ruby
      if messages && !messages.empty?
~~~

Reply:
I'm not sure that is the case, L233 messages is initialised to [], if there is an error with the payload, the check on 243 ```if !result.error?``` will fail logging an error and leaving messages as an empty array. Additionally, if the return from the api doesn't contain the `receivedMessages` key, L245 will leave messages as the result of JSON.parse(result.body). 

It may be worth a slight refactor of the branching logic here to reduce the number of duplicate checks made. 

I'll make these changes and push back shortly